src/core/Rule.ml

```diff
@@ -358,6 +358,23 @@ and extract_reduction = Separate | Concat [@@deriving show]
 and extract_transform = NoTransform | Unquote | ConcatJsonArray
 [@@deriving show]
 
+(*****************************************************************************)
+(* Secrets mode *)
+(*****************************************************************************)
+
+(*TODO: actualy figure this out*)
+type secrets_spec = {
+  (* secrets: *)
+  secrets : formula list;
+  (* request: *)
```

Comment:
better define intermediate structures for those, like request_spec, response_spec (or just request and response)

Reply:
Added `request_spec` && `response_spec`!